upgrade/v1beta1/upgrade.go

```diff
@@ -44,6 +44,21 @@ func UpgradeCRFromV1beta1ToV1beta2(config *restclient.Config, namespace string,
 	if err := upgradeNodes(namespace, lhClient); err != nil {
 		return err
 	}
+	if err := upgradeRecurringJobs(namespace, lhClient); err != nil {
+		return err
+	}
+	if err := upgradeBackups(namespace, lhClient); err != nil {
+		return err
+	}
+	if err := upgradeBackingImageDataSources(namespace, lhClient); err != nil {
+		return err
+	}
+	if err := upgradeBackingImageManagers(namespace, lhClient); err != nil {
+		return err
+	}
+	if err := upgradeBackingImages(namespace, lhClient); err != nil {
+		return err
+	}
```

Comment:
For these CRs, it's better to check if the objects are modified before updating them.

Reply:
Hmm, but how to check if the objects are modified?

Reply:
> Hmm, but how to check if the objects are modified?

- copy the existing CR before doing any operation on the CR
- run reflect.DeepEqual check before call the Update/UpdateStatus API